src/AS_02_IAB.cpp

```diff
@@ -272,7 +272,7 @@ AS_02::IAB::MXFWriter::Finalize() {
 
     byte_t clip_buffer[8] = { 0 };
 
-    ui64_t size = static_cast<ui64_t>(this->m_Writer->m_StreamOffset);
+    ui64_t size = static_cast<ui64_t>(this->m_Writer->m_StreamOffset) /* total size of the KLV */ - 24 /* K and L length */;
```

Comment:
Can the payload have a different offset, not just 24?

Reply:
24 is hard-coded at https://github.com/cinecert/asdcplib/blob/38daae206397d601738df407df15d1bf1caf01ae/src/AS_02_IAB.cpp#L180 (16 bytes for the UL and 8 bytes for the BER length)

This should probably be turned into a constant...

Reply:
Yes, those 24s are all over the place...

Reply:
@kblinova See revised source.